### PR TITLE
Why3 1.7.0

### PR DESCRIPTION
**Pull request by silene**

### CODE REVIEW

**Comment by silene** [comment]
None of the remaining failures are due to the added packages, so this is ready.

File: packages/why3-coq/why3-coq.1.7.0/opam [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   "--disable-frama-c"
+   "--disable-ide"
+   "--disable-js-of-ocaml"]
+  [make "-j%{jobs}%" "coq"]
+]
+
+install: [make "install-coq"]
+
+depends: [
+  "conf-autoconf" {build & dev}
```
Comment (haochenx) [inline packages/why3-coq/why3-coq.1.7.0/opam:40]: https://github.com/ocaml/opam-repository/pull/23843/files

see
```suggestion
  "conf-autoconf" {build | dev}
```

File: packages/why3-ide/why3-ide.1.7.0/opam [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   "--disable-re"
+   "--enable-ocamlfind"
+   "--enable-ide"]
+  [make "-j%{jobs}%" "ide"]
+]
+
+install: [make "install-ide"]
+
+depends: [
+  "conf-autoconf" {build & dev}
```
Comment (haochenx) [inline packages/why3-ide/why3-ide.1.7.0/opam:43]: https://github.com/ocaml/opam-repository/pull/23843/files

see
```suggestion
  "conf-autoconf" {build | dev}
```

File: packages/why3/why3.1.7.0/opam [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  [make "doc" "stdlibdoc" "apidoc"] {with-doc}
+]
+
+install: [
+  [make "install" "install-lib"]
+  [make "DOCDIR=%{_:doc}%" "install-doc"] {with-doc}
+]
+
+depends: [
+  "conf-autoconf" {build & dev}
```
Comment (haochenx) [inline packages/why3/why3.1.7.0/opam:56]: https://github.com/ocaml/opam-repository/pull/23843/files

see
```suggestion
  "conf-autoconf" {build | dev}
```

File: packages/why3/why3.1.7.0/opam [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  "mlmpfr"
+]
+
+conflicts: [
+  "why3-base"
+  "ocamlgraph" {< "1.8.2"}
+  "mlmpfr" {< "4.0.0"}
+  "base-effects"
+]
+
```
Comment (haochenx) [inline packages/why3/why3.1.7.0/opam:79]: It seems that cygwin patch is remove 1.6.0 → 1.7.0.

Is it planned to omit it for now and add it back after https://github.com/ocaml/opam-repository/pull/24841 getting merged?

**Comment by haochenx** [comment]
@silene it's probably a good idea to up stream https://github.com/ocaml/opam-repository/pull/23843

**Comment by silene** [comment]
The archive already contains a fully functional `configure` script; there is no reason to force the users to install `autoconf` to produce a new one.

**Comment by haochenx** [comment]
Then I guess maybe the dependency to conf-autoconf should be removed?

**Comment by silene** [comment]
No, it is needed for people who are directly building from the git repository, since we are not versioning the automatically generated  `configure` script, hence why the dependency on `conf-autoconf` is only there when Opam runs in `dev` mode.

**Comment by haochenx** [comment]
Ah. So does it mean patches like https://github.com/ocaml/opam-repository/pull/23843 is no longer needed for this new version?

**Comment by silene** [comment]
It was not needed either for the older releases, since the `configure` script has always been shipped in the archive. But for some unknown reason, in some very rare cases, the `configure` script was considered to be out-of-date, which caused Make to try to regenerate it from scratch. (A filesystem issue?) So, yes, depending on `conf-autoconf` does work around the issue, but it is way overkill. It is much cheaper to just ensure that `configure` is up-to-date by touching it just after extraction.

**Review by haochenx — APPROVED** [review]
All looks good to me. CI failures are unrelated as [mentioned by the author](https://github.com/ocaml/opam-repository/pull/24837#issuecomment-1826206343)

File: packages/why3-coq/why3-coq.1.7.0/opam [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  "deductive"
+  "program verification"
+  "formal specification"
+  "automated theorem prover"
+  "interactive theorem prover"
+]
+
+build: [
+  ["./autogen.sh"] {dev} # when pinning, there might be no configure file
+  ["touch" "configure"]
```
[the remark this answers is not in the document]
Comment (silene) [reply packages/why3-coq/why3-coq.1.7.0/opam:27]: I do not think it is worth the effort, now that Why3 1.7 is available.

**Pull request merged**